core/components/atoms/input/__tests__/Input.test.tsx

```diff
@@ -209,3 +209,773 @@ describe('Input component selection range is set correctly on input type change'
+      const { queryByTestId } = render(<Input name="test" value="" actionIcon={customActionIcon} />);
+
+      const actionButton = queryByTestId('DesignSystem-Input-ActionButton');
+      expect(actionButton).not.toBeInTheDocument();
+    });
+
+    it('prioritizes action icon over clear button', () => {
+      const onClearMock = jest.fn();
+      const customActionIcon = <Input.ActionButton name="custom" onClick={jest.fn()} />;
+      const { getByTestId, queryByTestId } = render(
+        <Input name="test" value="test value" onClear={onClearMock} actionIcon={customActionIcon} />
+      );
+
+      const actionButton = getByTestId('DesignSystem-Input-ActionButton');
+      const clearIcon = queryByTestId('DesignSystem-Input--closeIcon');
+
+      expect(actionButton).toBeInTheDocument();
+      expect(clearIcon).not.toBeInTheDocument();
+    });
+  });
+
+  describe('Info tooltip functionality', () => {
+    it('renders info tooltip when info prop is provided', () => {
+      const { container } = render(<Input name="test" info="This is helpful information" />);
+
+      const infoWrapper = container.querySelector('[tabIndex="0"]');
+      expect(infoWrapper).toBeInTheDocument();
+    });
+
+    it('does not render info tooltip when disabled', () => {
+      const { container } = render(<Input name="test" info="This is helpful information" disabled={true} />);
+
+      const infoWrapper = container.querySelector('[tabIndex="0"]');
+      expect(infoWrapper).not.toBeInTheDocument();
+    });
+  });
+
+  describe('Focus management', () => {
+    it('focuses input programmatically when needed', () => {
+      const { getByTestId } = render(<Input name="test" />);
+
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+
+      input.focus = jest.fn();
+      input.focus();
+
+      expect(input.focus).toHaveBeenCalled();
+      expect(input).toBeInTheDocument();
+    });
+
+    it('sets tabIndex to -1 when readOnly is true', () => {
+      const { getByTestId } = render(<Input name="test" readOnly={true} />);
+
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+      expect(input.tabIndex).toBe(-1);
+    });
+
+    it('does not set tabIndex when readOnly is false', () => {
+      const { getByTestId } = render(<Input name="test" readOnly={false} />);
+
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+      expect(input.tabIndex).not.toBe(-1);
+    });
+  });
+
+  describe('State management', () => {
+    it('updates isInputBlank state when wrapper loses focus', () => {
+      const { getByTestId } = render(<Input name="test" />);
+
+      const wrapper = getByTestId('DesignSystem-InputWrapper');
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+
+      fireEvent.change(input, { target: { value: 'test' } });
+
+      fireEvent.blur(wrapper);
+
+      expect(wrapper).toBeInTheDocument();
+    });
+  });
+
+  describe('Error handling', () => {
```

Comment:
Please write proper description for all the test cases. Also specify the **"Input Component"** in the description to clearly identify that this test case belongs to which component.